src/runtime/Story.svelte

```diff
@@ -106,17 +136,19 @@
 </script>
 
 {#if isCurrentlyViewed}
-  {#if children}
-    {@render children(renderer.args, renderer.storyContext)}
-  {:else if template}
+  {#if template && isSnippet(template)}
```

Comment:
we have 3 different props relevant here

template, children and component (via renderer.storyContext)

My understanding would be

1) if template exists, render it and pass children along
2) if component exists,  render it and pass children along
3) if children exists, render it

now this favors the component over a static story, the case where you have both component and children set is a bit of a conflict of interest, you have to take a guess if the user wants to render a static story overriding the default component or wants to pass the children prop into the default component.

Maybe this needs a <Story static> or <Story forwardChildren> prop to disambiguate?

Reply:
Let me see if I understand what you're proposing.

With the following case:

```svelte
<script module>
  import { defineMeta } from "@storybook/addon-svelte-csf";
  import Button from "./Button.svelte";
  <!--   👆 default component               👇 for the stories file -->
  const { Story } = defineMeta({ component: Button });
</script>

<Story name="Default">
  Click me
</Story>
```

The following piece:

```svelte
<Story name="Default">
  <span>Click me</span>
</Story>
```

Will produce:

```html
<Button>
  <span>Click me</span>
</Button>
```

Now, with the suggestion of adding a new `static` boolean flag prop to `Story` like this:

```svelte
<Story name="Default" static>
  <span>Click me</span>
</Story>
```

Will produce:

```html
<span>Click me</span>
```

Did I get this right?

Reply:
We continued this discussion on Storybook Discord and concluded by creating a new prop for this use case - `asChild`.